app/src/main/java/com/chouten/app/ui/views/search/SearchPageViewModel.kt

```diff
@@ -50,27 +51,90 @@ class SearchPageViewModel(
             field = value
             _searchQuery.value = value
             searchJob?.cancel()
-            searchJob = viewModelScope.launch {
-                _searchQuery.debounce(500).distinctUntilChanged()
-                    .collectLatest {
-                        isSearching = true
-                        if (it.isBlank()) {
-                            _searchResults.clear()
-                            isSearching = false
-                            return@collectLatest
+            searchJob =
+                    viewModelScope.launch {
+                        _searchQuery.debounce(500).distinctUntilChanged().collectLatest {
+                            isSearching = true
+                            if (it.isBlank()) {
+                                _searchResults.clear()
+                                isSearching = false
+                                return@collectLatest
+                            }
+                            println("Searching for $it")
+                            _searchQuery.value = it
+                            search(it)
                         }
-                        println("Searching for $it")
-                        _searchQuery.value = it
-                        search(it)
                     }
-            }
         }
 
     var previousSearchQuery by mutableStateOf("")
         private set // We don't want to be able to set this from outside the class.
 
     init {
         webview.initialize(context)
+        webview.dontCloseOnError()
+        webview.setCallback(this::callback)
+    }
+
+    fun getCode(): String{
+        val currentModule = ModuleLayer.selectedModule;
+        val subtype = currentModule?.subtypes?.getOrNull(0);
+        var code: String;
+
+        if(subtype == null){
+            throw Exception("Subtype not found");
+        }else{
+            val tempCode = currentModule.code?.get(subtype)?.search?.getOrNull(0)?.code;
+
+            if(tempCode == null){
```

Comment:
This could possibly be reduced. If the subtype is null, we can exit early by checking if there is actually a selected module in the first place.
`val currentModule = ModuleLayer.selectedModule ?: throw Exception("No module selected")`. We can then clean up the rest by doing `val subtype = currentModule?.subtypes?.getOrNull(0) ?: throw Exception("Subtype not found");`
Since we only actually instantiate `code` once, it could be a `val` rather than `var`. Also, that entire block can be reduced. WIth the previous change, the subtype check would be redundant. From there, we can just do `currentModule.code.get(subtype)?.search?.getOrNull(0)?.code ?: throw Exception("Code not found")`